src/GitHub.App/Services/TeamExplorerContext.cs

```diff
@@ -80,27 +84,47 @@ void Refresh()
                     if (newRepositoryPath != repositoryPath)
                     {
                         log.Debug("ActiveRepository changed to {CloneUrl} @ {Path}", repo?.CloneUrl, newRepositoryPath);
-                        ActiveRepository = repo;
+                        joinableTaskFactory.Run(async () =>
```

Comment:
Might it be worth adding a `RunOnMainThread()` extension method to do this, or is it better to make it explicit like this?

Reply:
When `Refresh` becomes `async` (from your other PR), all we'll need is `await joinableTaskFactory.SwitchToMainThreadAsync()` before the `ActiveRepository = repo`.